File: C/projects/CARDAMOM_MODELS/DALEC/DALEC_ALL/HYDROLOGY_MODULES/CONVERTERS/HYDROFUN_PSI2MOI.c
#pragma once

/*
Convert potential (MPa) to volumetric soil moisture (m3/m3)
Inputs:
psi_eff - effective water potential (MPa) of the liquid-saturated state (i.e. corrected for ice volume)
psi_porosity - porosity potential (MPa)
b - retention parameter
*/

double HYDROFUN_PSI2MOI(double psi, double psi_porosity, double b){

double moi = pow(psi/psi_porosity,(-1.0/b));
 
return moi;
}

File: PYTHON/projects/CARDAMOM_MODELS/DALEC/DALEC_ALL/HYDROLOGY_MODULES/CONVERTERS/HYDROFUN_PSI2MOI.py
import jax.numpy as jnp

def HYDROFUN_PSI2MOI(psi, psi_porosity, b):
    """Convert potential (MPa) to volumetric soil moisture (m3/m3)"""
    ratio = psi_porosity / jnp.minimum(psi, -1e-9)
    return jnp.power(ratio, 1.0 / b)
